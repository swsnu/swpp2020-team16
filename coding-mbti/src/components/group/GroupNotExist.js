import React from 'react';
import PropTypes from 'prop-types';

/* M-UIs */
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';

/* Components */
import GroupCreate from './GroupCreate';
import { ReactComponent as BackgroundImage2 } from './resources/teamwork.svg';

export default function GroupNotExist(props) {
  const { createGroup, error, isManager } = props;

  return (
    <>
      <div style={{ height: '25px' }} />
      <Container>
        <Grid container direction="row" justify="center" alignItems="center">
          <Grid item>
            <BackgroundImage2 height="300px" width="500px" alt="tuvieja" />
          </Grid>
        </Grid>
        <Grid container direction="row" justify="center" alignItems="center" textAlign="center">
          <Grid item>
            <Typography variant="h3" component="h3">
              You have no groups.
            </Typography>
          </Grid>
        </Grid>
        <Grid container direction="row" justify="center" alignItems="center" textAlign="center">
          <Grid item>
            <Typography variant="h3" component="h3">
              Create Your Own Group!
            </Typography>
          </Grid>
        </Grid>
        <div style={{ height: '25px' }} />
        <Grid container direction="row" justify="center" alignItems="center" textAlign="center">
          <Grid item>
            <GroupCreate createGroup={createGroup} error={error} isManager={isManager} />
          </Grid>
        </Grid>
        <div style={{ height: '25px' }} />
      </Container>
    </>
  );
}

GroupNotExist.propTypes = {
  createGroup: PropTypes.func.isRequired,
  error: PropTypes.object.isRequired,
  isManager: PropTypes.bool.isRequired,
};
