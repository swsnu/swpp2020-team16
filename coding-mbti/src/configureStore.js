import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  REGISTER,
  PAUSE,
  PERSIST,
  PURGE,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import reducer from './feature';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducer);

export default () => {
  const store = configureStore({
    reducer: persistedReducer,
    middleware: getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
  });
  const persistor = persistStore(store);
  return { store, persistor };
};
