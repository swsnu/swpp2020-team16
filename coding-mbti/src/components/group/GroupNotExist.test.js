import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { ThemeProvider, createMuiTheme } from '@material-ui/core';
import { transitions, positions, Provider as AlertProvider } from 'react-alert';
import AlertTemplate from 'react-alert-template-basic';
import GroupNotExist from './GroupNotExist';
import configureStore from '../../configureStore';

const theme = createMuiTheme();

const { store } = configureStore();
const options = {
    // you can also just use 'bottom center'
    position: positions.BOTTOM_CENTER,
    timeout: 5000,
    offset: '30px',
    // you can also just use 'scale'
    transition: transitions.SCALE,
};
describe('GroupNotExist', () => {
    let wrapper;
    const createGroup = jest.fn();

    it('render', () => {
        wrapper = mount(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <AlertProvider template={AlertTemplate} {...options}>
                        <GroupNotExist groups={{}} isManager gruopId={1} createGroup={createGroup} />
                    </AlertProvider>
                </ThemeProvider>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
