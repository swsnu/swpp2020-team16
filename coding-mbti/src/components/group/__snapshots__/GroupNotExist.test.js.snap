// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupNotExist render 1`] = `ReactWrapper {}`;
